### PR TITLE
api endpoint

### PR DESCRIPTION
**Pull request by martinkearn**
This PR adds the API endpoint as per the design as well as some notes on the code flow for when the relevant service are hooked up.

The branch names is not relevant to what the PR is for, but please ignore that.

Also removes redundant template files.

### CODE REVIEW

**Review by priyankasharma55 — APPROVED** [review]
Looks good.

File: src/MockClassifier.Api/Controllers/InstitutionController.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MockClassifier.Api.Models;
+
+namespace MockClassifier.Api.Controllers
+{
+    [Route("input-from-dmr/institution")]
```
Comment (decodingahmed) [inline src/MockClassifier.Api/Controllers/InstitutionController.cs:6]: Apologies, I missed this in the docs but I think this should be `/classify` (drop `input-from-dmr`). Currently posting to the `input-from-dmr/institution` resource implies (in REST terms) that you're the creating/posting a new institution.

File: src/MockClassifier.Api/Controllers/InstitutionController.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    {
+        /// <summary>
+        /// Processes an array of strings to identify corresponding ministries and issues call backs to DMR for each message.
+        /// </summary>
+        /// <param name="messages">Property which contains an array of strings representing a user message in each string</param>
+        /// <returns>An empty 202/Accepted result</returns>
+        [HttpPost]
+        public AcceptedResult Post([FromBody] MessagesInput messages)
+        {
+            var minsitries = new List<string>();
```
Comment (decodingahmed) [inline src/MockClassifier.Api/Controllers/InstitutionController.cs:18]:
```suggestion
            var ministries = new List<string>();
```

File: src/MockClassifier.Api/Controllers/InstitutionController.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+            // TO DO
+            // Enumerate each string in messages and check for
+            // 1) Does the string match one of the pre-defined phrases? If so, work out the corresponding ministry and put it into minsitries
+            // 2) Does the string include one of more of the pre-defined tokens? If so, get the corresponding minsitries and put them into minsitries
+
+            foreach (var minsitry in minsitries)
```
Comment (decodingahmed) [inline src/MockClassifier.Api/Controllers/InstitutionController.cs:25]:
```suggestion
            foreach (var ministry in ministries)
```

**Review by decodingahmed — CHANGES_REQUESTED** [review]
Sorry, I realise that this is merged already but I was halfway through my review.

**Pull request merged**